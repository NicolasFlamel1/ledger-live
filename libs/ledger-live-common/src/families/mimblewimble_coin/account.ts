import type { Operation } from "@ledgerhq/types-live";
import Identifier from "./api/identifier";
import type { MimbleWimbleCoinOperationExtra, MimbleWimbleCoinOperationExtraRaw } from "./types";

const formatOperationSpecifics = (operation: Operation): string => {
  const { outputCommitment } = operation.extra as MimbleWimbleCoinOperationExtra;
  return outputCommitment ? `\n  Output Commitment: ${outputCommitment.toString("hex")}` : "";
};

export const fromOperationExtraRaw = (
  extraRaw: MimbleWimbleCoinOperationExtraRaw,
): MimbleWimbleCoinOperationExtra => {
  const extra: MimbleWimbleCoinOperationExtra = {};
  if (extraRaw.outputCommitment !== undefined) {
    extra.outputCommitment = Buffer.from(extraRaw.outputCommitment, "hex");
  }
  if (extraRaw.identifier !== undefined) {
    extra.identifier = new Identifier(Buffer.from(extraRaw.identifier, "hex"));
  }
  if (extraRaw.switchType !== undefined) {
    extra.switchType = extraRaw.switchType;
  }
  if (extraRaw.spent !== undefined) {
    extra.spent = extraRaw.spent;
  }
  if (extraRaw.kernelExcess !== undefined) {
    extra.kernelExcess = extraRaw.kernelExcess ? Buffer.from(extraRaw.kernelExcess, "hex") : null;
  }
  if (extraRaw.kernelOffset !== undefined) {
    extra.kernelOffset = extraRaw.kernelOffset ? Buffer.from(extraRaw.kernelOffset, "hex") : null;
  }
  if (extraRaw.recipientPaymentProofSignature !== undefined) {
    extra.recipientPaymentProofSignature = extraRaw.recipientPaymentProofSignature
      ? Buffer.from(extraRaw.recipientPaymentProofSignature, "hex")
      : null;
  }
  return extra;
};

export const toOperationExtraRaw = (
  extra: MimbleWimbleCoinOperationExtra,
): MimbleWimbleCoinOperationExtraRaw => {
  const extraRaw: MimbleWimbleCoinOperationExtraRaw = {};
  if (extra.outputCommitment !== undefined) {
    extraRaw.outputCommitment = extra.outputCommitment.toString("hex");
  }
  if (extra.identifier !== undefined) {
    extraRaw.identifier = extra.identifier.serialize().toString("hex");
  }
  if (extra.switchType !== undefined) {
    extraRaw.switchType = extra.switchType;
  }
  if (extra.spent !== undefined) {
    extraRaw.spent = extra.spent;
  }
  if (extra.kernelExcess !== undefined) {
    extraRaw.kernelExcess = extra.kernelExcess ? extra.kernelExcess.toString("hex") : null;
  }
  if (extra.kernelOffset !== undefined) {
    extraRaw.kernelOffset = extra.kernelOffset ? extra.kernelOffset.toString("hex") : null;
  }
  if (extra.recipientPaymentProofSignature !== undefined) {
    extraRaw.recipientPaymentProofSignature = extra.recipientPaymentProofSignature
      ? extra.recipientPaymentProofSignature.toString("hex")
      : null;
  }
  return extraRaw;
};

export default {
  formatOperationSpecifics,
  fromOperationExtraRaw,
  toOperationExtraRaw,
};
