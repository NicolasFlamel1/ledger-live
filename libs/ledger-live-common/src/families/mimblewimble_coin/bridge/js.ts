import type { AccountBridge, CurrencyBridge } from "@ledgerhq/types-live";
import type { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
import type { Transaction } from "../types";
import getTransactionStatus from "../js-getTransactionStatus";
import estimateMaxSpendable from "../js-estimateMaxSpendable";
import signOperation from "../js-signOperation";
import broadcast from "../js-broadcast";
import { scanAccounts, sync } from "../js-synchronisation";
import { createTransaction, updateTransaction, prepareTransaction } from "../js-transaction";
import { makeAccountBridgeReceive } from "../../../bridge/jsHelpers";

const getPreloadStrategy = (
  currency: CryptoCurrency
): any => {};

const preload = async (): Promise<any> => {};

const hydrate = (): void => {};

const currencyBridge: CurrencyBridge = {
  getPreloadStrategy,
  preload,
  hydrate,
  scanAccounts
};

const receive = makeAccountBridgeReceive();

const accountBridge: AccountBridge<Transaction> = {
  estimateMaxSpendable,
  createTransaction,
  updateTransaction,
  getTransactionStatus,
  prepareTransaction,
  sync,
  receive,
  signOperation,
  broadcast
};

export default {
  currencyBridge,
  accountBridge
};
