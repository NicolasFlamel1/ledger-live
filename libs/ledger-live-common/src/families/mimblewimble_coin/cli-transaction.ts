import type { AccountLike } from "@ledgerhq/types-live";
import type { Transaction } from "../../generated/types";
import invariant from "invariant";
import flatMap from "lodash/flatMap";

const inferTransactions = (
  transactions: {
    account: AccountLike,
    transaction: Transaction
  }[]
): Transaction[] => {
  return flatMap(transactions, ({
    transaction
  }: {
    transaction: Transaction;
  }): Transaction => {
    invariant(transaction.family === "mimblewimble_coin", "mimblewimble_coin family");
    return {
      ...transaction,
      family: "mimblewimble_coin",
      sendAsFile: false,
      height: undefined,
      id: undefined,
      offset: undefined,
      proof: undefined,
      encryptedSecretNonce: undefined,
      transactionResponse: undefined
    };
  });
};

export default {
  inferTransactions
};
