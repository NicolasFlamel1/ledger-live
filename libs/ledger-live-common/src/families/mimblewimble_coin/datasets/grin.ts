import type { CurrenciesData } from "@ledgerhq/types-live";
import type { Transaction, MimbleWimbleCoinAccountRaw } from "../types";
import { fromTransactionRaw } from "../transaction";
import { AmountRequired, NotEnoughBalance, RecipientRequired, InvalidAddress } from "@ledgerhq/errors";
import scanAccounts1 from "./grin.scanAccounts.1";

export default {
  scanAccounts: [scanAccounts1],
  accounts: [{
    raw: {
      id: "js:2:grin:59a666ad7a04a9c397924d14b0e1faba023559b4c23c102df59ad0f20773a2b189bc0acacbf385a1d8997a76ac2b3b7e0332187b85b24653702eec838e32d3b8:",
      seedIdentifier: "",
      xpub: "59a666ad7a04a9c397924d14b0e1faba023559b4c23c102df59ad0f20773a2b189bc0acacbf385a1d8997a76ac2b3b7e0332187b85b24653702eec838e32d3b8",
      derivationMode: "",
      index: 0,
      freshAddress: "grin1pe6nvyy4n2nqh73l8kr80zqj5uf4cyqdu4jelqhxmmmjufw8g33q42r76l",
      freshAddressPath: "44'/592'/0'/0/0",
      freshAddresses: [{
        address: "grin1pe6nvyy4n2nqh73l8kr80zqj5uf4cyqdu4jelqhxmmmjufw8g33q42r76l",
        derivationPath: "44'/592'/0'/0/0"
      }],
      name: "Grin 1",
      balance: "0",
      spendableBalance: "0",
      blockHeight: 0,
      operationsCount: 0,
      currencyId: "grin",
      operations: [],
      pendingOperations: [],
      unitMagnitude: 9,
      lastSyncDate: "",
      mimbleWimbleCoinResources: {
        rootPublicKey: "03efd2936a0ec29e49aff8ef7d6d08c42b8b39d5c632ba240b73f98a8f89a79a8a",
        recentHeights: [],
        nextIdentifier: "0300000000000000000000000000000000",
        nextTransactionSequenceNumber: 0
      }
    } as MimbleWimbleCoinAccountRaw,
    transactions: [{
      name: "Amount required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          amount: new AmountRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Not enough balance",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "1",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          amount: new NotEnoughBalance()
        },
        warnings: {}
      }
    },
    {
      name: "Recipient required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          recipient: new RecipientRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Invalid address",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          recipient: new InvalidAddress()
        },
        warnings: {}
      }
    }]
  }]
} as CurrenciesData<Transaction>;
