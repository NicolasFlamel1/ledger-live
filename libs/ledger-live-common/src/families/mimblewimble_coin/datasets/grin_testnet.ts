import type { CurrenciesData } from "@ledgerhq/types-live";
import type { Transaction, MimbleWimbleCoinAccountRaw } from "../types";
import { fromTransactionRaw } from "../transaction";
import { AmountRequired, NotEnoughBalance, RecipientRequired, InvalidAddress } from "@ledgerhq/errors";
import scanAccounts1 from "./grin_testnet.scanAccounts.1";

export default {
  scanAccounts: [scanAccounts1],
  accounts: [{
    raw: {
      id: "js:2:grin_testnet:2d4844e2f8aa56a6a41c56fc89a685a5b65a53ee88e57d90d9da35acb57860c0c05f33206382ccbcd04094d5c7651016b4e3997bdf4ff97e4fbe41178f0376fe:",
      seedIdentifier: "",
      xpub: "2d4844e2f8aa56a6a41c56fc89a685a5b65a53ee88e57d90d9da35acb57860c0c05f33206382ccbcd04094d5c7651016b4e3997bdf4ff97e4fbe41178f0376fe",
      derivationMode: "",
      index: 0,
      freshAddress: "tgrin1txasfcykjh7vk47tadwy24x2wqhexkjk3e93nx7kuah8q87arlqsa3cnlw",
      freshAddressPath: "44'/1'/0'/0/0",
      freshAddresses: [{
        address: "tgrin1txasfcykjh7vk47tadwy24x2wqhexkjk3e93nx7kuah8q87arlqsa3cnlw",
        derivationPath: "44'/1'/0'/0/0"
      }],
      name: "Grin Testnet 1",
      balance: "0",
      spendableBalance: "0",
      blockHeight: 0,
      operationsCount: 0,
      currencyId: "grin_testnet",
      operations: [],
      pendingOperations: [],
      unitMagnitude: 9,
      lastSyncDate: "",
      mimbleWimbleCoinResources: {
        rootPublicKey: "036bf8b1016cfb395b5ef06a503c7c06742685ed0e8214bc02f6d6de44acc5f8e1",
        recentHeights: [],
        nextIdentifier: "0300000000000000000000000000000000",
        nextTransactionSequenceNumber: 0
      }
    } as MimbleWimbleCoinAccountRaw,
    transactions: [{
      name: "Amount required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        address: undefined,
        identifier: undefined,
        freshAddress: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "500000",
        networkInfo: {}
      }),
      expectedStatus: {
        errors: {
          amount: new AmountRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Not enough balance",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "1",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        address: undefined,
        identifier: undefined,
        freshAddress: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "500000",
        networkInfo: {}
      }),
      expectedStatus: {
        errors: {
          amount: new NotEnoughBalance()
        },
        warnings: {}
      }
    },
    {
      name: "Recipient required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        address: undefined,
        identifier: undefined,
        freshAddress: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "500000",
        networkInfo: {}
      }),
      expectedStatus: {
        errors: {
          recipient: new RecipientRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Invalid address",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        address: undefined,
        identifier: undefined,
        freshAddress: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "500000",
        networkInfo: {}
      }),
      expectedStatus: {
        errors: {
          recipient: new InvalidAddress()
        },
        warnings: {}
      }
    }]
  }]
} as CurrenciesData<Transaction>;
