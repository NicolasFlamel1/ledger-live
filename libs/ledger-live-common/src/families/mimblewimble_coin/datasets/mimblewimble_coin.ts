import type { CurrenciesData } from "@ledgerhq/types-live";
import type { Transaction, MimbleWimbleCoinAccountRaw } from "../types";
import { fromTransactionRaw } from "../transaction";
import { AmountRequired, NotEnoughBalance, RecipientRequired, InvalidAddress } from "@ledgerhq/errors";
import scanAccounts1 from "./mimblewimble_coin.scanAccounts.1";

export default {
  scanAccounts: [scanAccounts1],
  accounts: [{
    raw: {
      id: "js:2:mimblewimble_coin:096fe43a17434bb1d42a09142faeb5a5cf5a28f25c716e9b64b702a6e67dd5684c98d02fef8a7c969856225caef3e2c8b4b1a3a2b0dc2ef432ea36561e75fdd1:",
      seedIdentifier: "",
      xpub: "096fe43a17434bb1d42a09142faeb5a5cf5a28f25c716e9b64b702a6e67dd5684c98d02fef8a7c969856225caef3e2c8b4b1a3a2b0dc2ef432ea36561e75fdd1",
      derivationMode: "",
      index: 0,
      freshAddress: "htpob76thyhuc7zcfu3fm5rvii6um3v5vb7qtjubgqoryz6j6lb4txqd",
      freshAddressPath: "44'/593'/0'/0/0",
      freshAddresses: [{
        address: "htpob76thyhuc7zcfu3fm5rvii6um3v5vb7qtjubgqoryz6j6lb4txqd",
        derivationPath: "44'/593'/0'/0/0"
      }],
      name: "MimbleWimble Coin 1",
      balance: "0",
      spendableBalance: "0",
      blockHeight: 0,
      operationsCount: 0,
      currencyId: "mimblewimble_coin",
      operations: [],
      pendingOperations: [],
      unitMagnitude: 9,
      lastSyncDate: "",
      mimbleWimbleCoinResources: {
        rootPublicKey: "031d11847503f131f736448684c97fb743ea6324b1e02e4a963c0016e3b4f50674",
        recentHeights: [],
        nextIdentifier: "0300000000000000000000000000000000",
        nextTransactionSequenceNumber: 0
      }
    } as MimbleWimbleCoinAccountRaw,
    transactions: [{
      name: "Amount required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "1000000"
      }),
      expectedStatus: {
        errors: {
          amount: new AmountRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Not enough balance",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "1",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "1000000"
      }),
      expectedStatus: {
        errors: {
          amount: new NotEnoughBalance()
        },
        warnings: {}
      }
    },
    {
      name: "Recipient required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "1000000"
      }),
      expectedStatus: {
        errors: {
          recipient: new RecipientRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Invalid address",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined,
        useDefaultBaseFee: true,
        baseFee: "1000000"
      }),
      expectedStatus: {
        errors: {
          recipient: new InvalidAddress()
        },
        warnings: {}
      }
    }]
  }]
} as CurrenciesData<Transaction>;
