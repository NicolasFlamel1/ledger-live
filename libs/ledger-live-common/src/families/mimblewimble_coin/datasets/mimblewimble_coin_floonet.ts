import type { CurrenciesData } from "@ledgerhq/types-live";
import type { Transaction, MimbleWimbleCoinAccountRaw } from "../types";
import { fromTransactionRaw } from "../transaction";
import { AmountRequired, NotEnoughBalance, RecipientRequired, InvalidAddress } from "@ledgerhq/errors";
import scanAccounts1 from "./mimblewimble_coin_floonet.scanAccounts.1";

export default {
  scanAccounts: [scanAccounts1],
  accounts: [{
    raw: {
      id: "js:2:mimblewimble_coin_floonet:fe30f5f7b405c4475289105d1326dded1b8d5a6b1315da4f70ef69a00d28609f18cafa041d44f3388162ef452f53bf1ba5b08f449c95f96362896f1124b207b6:",
      seedIdentifier: "",
      xpub: "fe30f5f7b405c4475289105d1326dded1b8d5a6b1315da4f70ef69a00d28609f18cafa041d44f3388162ef452f53bf1ba5b08f449c95f96362896f1124b207b6",
      derivationMode: "",
      index: 0,
      freshAddress: "7rgrlqdcygvsyjlrawpidkyy4bys67t343kh7knshea2gyld6gz67fid",
      freshAddressPath: "44'/1'/0'/0/0",
      freshAddresses: [{
        address: "7rgrlqdcygvsyjlrawpidkyy4bys67t343kh7knshea2gyld6gz67fid",
        derivationPath: "44'/1'/0'/0/0"
      }],
      name: "MimbleWimble Coin Floonet 1",
      balance: "0",
      spendableBalance: "0",
      blockHeight: 0,
      operationsCount: 0,
      currencyId: "mimblewimble_coin_floonet",
      operations: [],
      pendingOperations: [],
      unitMagnitude: 9,
      lastSyncDate: "",
      mimbleWimbleCoinResources: {
        rootPublicKey: "035c1b539dec12322947c8dc091f2303ed7552d138790147677d87c3f1679eb192",
        recentHeights: [],
        nextIdentifier: "0300000000000000000000000000000000",
        nextTransactionSequenceNumber: 0
      }
    } as MimbleWimbleCoinAccountRaw,
    transactions: [{
      name: "Amount required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          amount: new AmountRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Not enough balance",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "1",
        recipient: "http://localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          amount: new NotEnoughBalance()
        },
        warnings: {}
      }
    },
    {
      name: "Recipient required",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          recipient: new RecipientRequired()
        },
        warnings: {}
      }
    },
    {
      name: "Invalid address",
      transaction: fromTransactionRaw({
        family: "mimblewimble_coin",
        amount: "0",
        recipient: "localhost",
        useAllAmount: false,
        sendAsFile: false,
        height: undefined,
        id: undefined,
        offset: undefined,
        proof: undefined,
        encryptedSecretNonce: undefined,
        transactionResponse: undefined
      }),
      expectedStatus: {
        errors: {
          recipient: new InvalidAddress()
        },
        warnings: {}
      }
    }]
  }]
} as CurrenciesData<Transaction>;
