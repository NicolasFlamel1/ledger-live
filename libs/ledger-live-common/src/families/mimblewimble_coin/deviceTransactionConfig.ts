import type { Transaction, TransactionStatus } from "./types";
import type { DeviceTransactionField } from "../../transaction";
import { MimbleWimbleCoinTransactionWontHavePaymentProof } from "./errors";

export default (
  {
    transaction,
    status
  }: {
    transaction: Transaction;
    status: TransactionStatus;
  }
): DeviceTransactionField[] => {
  const fields: DeviceTransactionField[] = [];
  fields.push({
    type: "amount",
    label: "Amount"
  });
  fields.push({
    type: "fees",
    label: "Fee"
  });
  fields.push({
    type: "text",
    label: "Kernel Features",
    value: "Plain"
  });
  fields.push({
    type: "text",
    label: "Recipient Payment Proof Address",
    value: (status.warnings.recipient instanceof MimbleWimbleCoinTransactionWontHavePaymentProof) ? "N/A" : transaction.recipient.trim()
  });
  return fields;
}
