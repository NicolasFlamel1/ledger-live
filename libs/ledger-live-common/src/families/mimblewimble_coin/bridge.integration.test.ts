import "../../__tests__/test-helpers/setup";
import { testBridge } from "../../__tests__/test-helpers/bridge";
import type { DatasetTest } from "@ledgerhq/types-live";
import type { Transaction } from "./types";
import mimblewimble_coin from "./datasets/mimblewimble_coin";
import mimblewimble_coin_floonet from "./datasets/mimblewimble_coin_floonet";
import grin from "./datasets/grin";
import grin_testnet from "./datasets/grin_testnet";

const dataset: DatasetTest<Transaction> = {
  implementations: ["js"],
  currencies: {
    mimblewimble_coin,
    mimblewimble_coin_floonet,
    grin,
    grin_testnet
  }
};

testBridge(dataset);
