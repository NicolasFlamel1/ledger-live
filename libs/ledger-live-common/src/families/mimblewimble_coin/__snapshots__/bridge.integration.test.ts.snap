// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grin currency bridge scanAccounts grin seed 1 1`] = `
Array [
  Object {
    "balance": "0",
    "currencyId": "grin",
    "derivationMode": "",
    "freshAddress": "grin1pe6nvyy4n2nqh73l8kr80zqj5uf4cyqdu4jelqhxmmmjufw8g33q42r76l",
    "freshAddressPath": "44'/592'/0'/0/0",
    "freshAddresses": Array [
      Object {
        "address": "grin1pe6nvyy4n2nqh73l8kr80zqj5uf4cyqdu4jelqhxmmmjufw8g33q42r76l",
        "derivationPath": "44'/592'/0'/0/0",
      },
    ],
    "id": "js:2:grin:59a666ad7a04a9c397924d14b0e1faba023559b4c23c102df59ad0f20773a2b189bc0acacbf385a1d8997a76ac2b3b7e0332187b85b24653702eec838e32d3b8:",
    "index": 0,
    "mimbleWimbleCoinResources": Object {
      "nextIdentifier": "0300000000000000000000000000000000",
      "recentHeights": Array [],
      "rootPublicKey": "03efd2936a0ec29e49aff8ef7d6d08c42b8b39d5c632ba240b73f98a8f89a79a8a",
      "nextTransactionSequenceNumber": 0,
    },
    "name": "Grin 1",
    "nfts": undefined,
    "operationsCount": 0,
    "pendingOperations": Array [],
    "seedIdentifier": undefined,
    "spendableBalance": "0",
    "starred": false,
    "swapHistory": Array [],
    "syncHash": undefined,
    "unitMagnitude": 9,
    "used": false,
    "xpub": "59a666ad7a04a9c397924d14b0e1faba023559b4c23c102df59ad0f20773a2b189bc0acacbf385a1d8997a76ac2b3b7e0332187b85b24653702eec838e32d3b8",
  },
]
`;

exports[`grin currency bridge scanAccounts grin seed 1 2`] = `
Array [
  Array [],
]
`;

exports[`grin_testnet currency bridge scanAccounts grin_testnet seed 1 1`] = `
Array [
  Object {
    "balance": "0",
    "currencyId": "grin_testnet",
    "derivationMode": "",
    "freshAddress": "tgrin1txasfcykjh7vk47tadwy24x2wqhexkjk3e93nx7kuah8q87arlqsa3cnlw",
    "freshAddressPath": "44'/1'/0'/0/0",
    "freshAddresses": Array [
      Object {
        "address": "tgrin1txasfcykjh7vk47tadwy24x2wqhexkjk3e93nx7kuah8q87arlqsa3cnlw",
        "derivationPath": "44'/1'/0'/0/0",
      },
    ],
    "id": "js:2:grin_testnet:2d4844e2f8aa56a6a41c56fc89a685a5b65a53ee88e57d90d9da35acb57860c0c05f33206382ccbcd04094d5c7651016b4e3997bdf4ff97e4fbe41178f0376fe:",
    "index": 0,
    "mimbleWimbleCoinResources": Object {
      "nextIdentifier": "0300000000000000000000000000000000",
      "recentHeights": Array [],
      "rootPublicKey": "036bf8b1016cfb395b5ef06a503c7c06742685ed0e8214bc02f6d6de44acc5f8e1",
      "nextTransactionSequenceNumber": 0,
    },
    "name": "Grin Testnet 1",
    "nfts": undefined,
    "operationsCount": 0,
    "pendingOperations": Array [],
    "seedIdentifier": undefined,
    "spendableBalance": "0",
    "starred": false,
    "swapHistory": Array [],
    "syncHash": undefined,
    "unitMagnitude": 9,
    "used": false,
    "xpub": "2d4844e2f8aa56a6a41c56fc89a685a5b65a53ee88e57d90d9da35acb57860c0c05f33206382ccbcd04094d5c7651016b4e3997bdf4ff97e4fbe41178f0376fe",
  },
]
`;

exports[`grin_testnet currency bridge scanAccounts grin_testnet seed 1 2`] = `
Array [
  Array [],
]
`;

exports[`mimblewimble_coin currency bridge scanAccounts mimblewimble_coin seed 1 1`] = `
Array [
  Object {
    "balance": "0",
    "currencyId": "mimblewimble_coin",
    "derivationMode": "",
    "freshAddress": "htpob76thyhuc7zcfu3fm5rvii6um3v5vb7qtjubgqoryz6j6lb4txqd",
    "freshAddressPath": "44'/593'/0'/0/0",
    "freshAddresses": Array [
      Object {
        "address": "htpob76thyhuc7zcfu3fm5rvii6um3v5vb7qtjubgqoryz6j6lb4txqd",
        "derivationPath": "44'/593'/0'/0/0",
      },
    ],
    "id": "js:2:mimblewimble_coin:096fe43a17434bb1d42a09142faeb5a5cf5a28f25c716e9b64b702a6e67dd5684c98d02fef8a7c969856225caef3e2c8b4b1a3a2b0dc2ef432ea36561e75fdd1:",
    "index": 0,
    "mimbleWimbleCoinResources": Object {
      "nextIdentifier": "0300000000000000000000000000000000",
      "recentHeights": Array [],
      "rootPublicKey": "031d11847503f131f736448684c97fb743ea6324b1e02e4a963c0016e3b4f50674",
      "nextTransactionSequenceNumber": 0,
    },
    "name": "MimbleWimble Coin 1",
    "nfts": undefined,
    "operationsCount": 0,
    "pendingOperations": Array [],
    "seedIdentifier": undefined,
    "spendableBalance": "0",
    "starred": false,
    "swapHistory": Array [],
    "syncHash": undefined,
    "unitMagnitude": 9,
    "used": false,
    "xpub": "096fe43a17434bb1d42a09142faeb5a5cf5a28f25c716e9b64b702a6e67dd5684c98d02fef8a7c969856225caef3e2c8b4b1a3a2b0dc2ef432ea36561e75fdd1",
  },
]
`;

exports[`mimblewimble_coin currency bridge scanAccounts mimblewimble_coin seed 1 2`] = `
Array [
  Array [],
]
`;

exports[`mimblewimble_coin_floonet currency bridge scanAccounts mimblewimble_coin_floonet seed 1 1`] = `
Array [
  Object {
    "balance": "0",
    "currencyId": "mimblewimble_coin_floonet",
    "derivationMode": "",
    "freshAddress": "7rgrlqdcygvsyjlrawpidkyy4bys67t343kh7knshea2gyld6gz67fid",
    "freshAddressPath": "44'/1'/0'/0/0",
    "freshAddresses": Array [
      Object {
        "address": "7rgrlqdcygvsyjlrawpidkyy4bys67t343kh7knshea2gyld6gz67fid",
        "derivationPath": "44'/1'/0'/0/0",
      },
    ],
    "id": "js:2:mimblewimble_coin_floonet:fe30f5f7b405c4475289105d1326dded1b8d5a6b1315da4f70ef69a00d28609f18cafa041d44f3388162ef452f53bf1ba5b08f449c95f96362896f1124b207b6:",
    "index": 0,
    "mimbleWimbleCoinResources": Object {
      "nextIdentifier": "0300000000000000000000000000000000",
      "recentHeights": Array [],
      "rootPublicKey": "035c1b539dec12322947c8dc091f2303ed7552d138790147677d87c3f1679eb192",
      "nextTransactionSequenceNumber": 0,
    },
    "name": "MimbleWimble Coin Floonet 1",
    "nfts": undefined,
    "operationsCount": 0,
    "pendingOperations": Array [],
    "seedIdentifier": undefined,
    "spendableBalance": "0",
    "starred": false,
    "swapHistory": Array [],
    "syncHash": undefined,
    "unitMagnitude": 9,
    "used": false,
    "xpub": "fe30f5f7b405c4475289105d1326dded1b8d5a6b1315da4f70ef69a00d28609f18cafa041d44f3388162ef452f53bf1ba5b08f449c95f96362896f1124b207b6",
  },
]
`;

exports[`mimblewimble_coin_floonet currency bridge scanAccounts mimblewimble_coin_floonet seed 1 2`] = `
Array [
  Array [],
]
`;
